### PR TITLE
[WIP - do not review] Add CTA button to directly take user to their state page

### PR DESCRIPTION
**Pull request by rachelrf**
This PR does a couple of things:

1. States pulse slightly when the user is not hovering, to make it clear that the map is interactive. This might conflict slightly with the changes to the map colors, so happy to merge these changes after.

2. Adds a "See my state" button. When tapped, it prompts the user for their browser location, then reverse geocodes for their US state using [this api](https://www.bigdatacloud.com/geocoding-apis/free-reverse-geocode-to-city-api) which is free and unlimited for client-side use. I think this is important functionality, because it is difficult to tap on the smaller states on mobile. It's also a nice and clear call to action for people that won't read the instructional text. 

Here is what it looks like on desktop and mobile:  

This change: desktop
Screen Shot 2020-03-22 at 10 30 30 PM
  

This change: mobile
Screen Shot 2020-03-22 at 10 30 23 PM
  

The style guide has black buttons, but I wanted to make the primary CTA inviting so I made it blue. Happy to change that if anyone has design guidance. 

Style guide button: desktop
Screen Shot 2020-03-22 at 10 31 28 PM

### CODE REVIEW

File: src/components/Map/Map.js [hunk 1]
```diff
@@ -1,31 +1,151 @@
...
+      clearTimeout(rouletteAnimationTimer);
+    }
+  }, [])
+
+  if (redirectTarget) {
+    return <Redirect push to={redirectTarget} />;
+  }
+
+  /* mandatory */
+  let mapHandler = event => {
```
Comment (zjrosen1) [inline src/components/Map/Map.js]: this can be removed

File: src/components/Map/Map.js [hunk 2]
```diff
@@ -1,31 +1,151 @@
...
+          setGeolocatedStateCode(stateCode);
+          resolve(stateCode);
+        })
+        .catch((error) => {
+          setSeeMyStateButtonHidden(true);
+        });
+    });
+  }
+
+  //function getClientLocation(callback) {
```
Comment (zjrosen1) [inline src/components/Map/Map.js]: remove

File: src/components/Map/Map.js (hunk 1 above)
Comment (zjrosen1) [reply src/components/Map/Map.js]: or it was removed and looks like you added it back?

**Comment by rachelrf** [comment]
Some changes in this commit:

1. Only expose the "See your state" functionality on mobile.

2. "See your state" is a link rather than a button

3. Removed the functionality to make states pulse when the user is not interacting with the map. This is too busy with the additional colors.

Screen Shot 2020-03-24 at 11 17 30 PM

**Comment by igorkofman** [comment]
Hey Rachel - this is really *nice* but we're in the process of rolling out
a search dropdown based interaction to solve the mobile navigation problem
while also allowing for navigating to a given city / county (once that's
available).

It could be really cool to integrate your idea by offering a "current
location" row in the dropdown.  Could you connect with @Jeremy Lubin
<lubin.jeremy@gmail.com> and see if that makes sense / design it out?

**Comment by rachelrf** [comment]
@igorkofman sg, I'll talk to Jeremy

**Pull request closed without merging**